api/controllers/v0.2/createPatron.js

```diff
@@ -135,25 +216,42 @@ function createPatron(req, res) {
         collectErrorResponseData(
           null,
           'invalid-request',
-          'Missing required patron information.',
+          'Required patron information is missing.',
           null,
           debugMessage // eslint-disable-line comma-dangle
         ) // eslint-disable-line comma-dangle
       ));
-
-    return;
+    return false;
   }
+  return true;
+}
+
+/**
+ * createPatron(req, res)
+ * The callback for the route "/patrons".
+ * It will fire a POST request to the ILS API for creating a new patron.
+ *
+ * @param {HTTP request} req
+ * @param {HTTP response} res
+ */
+function createPatron(req, res) {
+  if (validateEnvironmentAndRequest(req, res) === false) { return false; }
 
   ilsClientKey = ilsClientKey || awsDecrypt.decryptKMS(process.env.ILS_CLIENT_KEY);
   ilsClientPassword = ilsClientPassword || awsDecrypt.decryptKMS(process.env.ILS_CLIENT_SECRET);
 
+  // eslint-disable-next-line max-len
+  // Delete pcode4 data to prevent an error on Sierra TODO: remove this line once pcode4 is accepted; also remove related line in v0.2 modelStreamPatron.js
+  if (req.body && req.body.patronCodes) {
+    delete req.body.patronCodes.pcode4;
+  }
+
   Promise.all([ilsClientKey, ilsClientPassword]).then((values, reject) => {
     [ilsClientKey, ilsClientPassword] = values;
 
     const username = ilsClientKey;
     const password = ilsClientPassword;
     const basicAuth = `Basic ${base64(`${username}:${password}`)}`;
-
     axios.post(process.env.ILS_CREATE_TOKEN_URL, {}, {
```

Comment:
This may be beyond the scope of this work, but seems a shame to fetch a new token for every request. Ideally we'd use previously fetched tokens until they expire (and then refresh them). (But again, this may be beyond the scope of this work)